src/contextprops/index.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Loading, reducer as loadingReducer} from '../loading';
+import {contextProp} from '../context';
+
+/**
+ * Defines whether a DOM subtree can be currently seen by the user. A subtree
+ * can be not renderable due `display: none`, or `hidden` attribute, unslotted
+ * in shadow DOM, or because a parent component knows for certain that this
+ * subtree cannot be seen by the user without some active interraction.
+ *
+ * Default is `true`.
+ *
+ * @const {!ContextProp<boolean>}
+ */
+const CanRender = contextProp('CanRender', {
+  defaultValue: true,
+  recursive: (inputs) => inputs.reduce(andReducer),
+  compute: (contextNode, inputs, parentValue) =>
+    (parentValue && inputs.reduce(andReducer, true)) || false,
```

Comment:
Because `recursive` is a function, when it returns `false`, the parent is not computed and its value is `undefined`. Thus `undefined && inputs.reduce() === undefined`, so the `||false` is necessary to turn it to the strict `{boolean}`.